cmd/keymaster/main_test.go

```diff
@@ -275,3 +275,13 @@ func TestInsertSSHCertIntoAgentORWriteToFilesystem(t *testing.T) {
 	// TODO: on linux/macos create agent + unix socket and pass that
 
 }
+
+func TestMainPrintVersion(t *testing.T) {
+	os.Args = []string{"cmd", "-version"}
+	done := make(chan struct{})
+	go func() {
+		main()
+		close(done)
```

Comment:
This is not really a good test as you are not even checking for the output of the call or its side-effects. Testing main IS complicated. And I think the way that is needed is to make a new : mainWithError(logger  log.Logger) error. That basically does everything except that it no longer calls fatal, but returns and error. And the main will just call this mainWithError with the initialized logger. Now we can start testing mainWithError. And check for the return types and the sideEffects.

Not for today (unless you want too).

Reply:
@cviecco I added the test, pls check